### PR TITLE
LOG-5725: Refactor multilineexception functional test

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Enables and refactors the functional multiline_exception_test to use the obs API
* Refactors the multiline exception logic into a filter package
* Removes redundant tests
* Removes test that provide no additional coverage
* Enables previously skipped tests where applicable

### Links
https://issues.redhat.com/browse/LOG-5725

cc @Clee2691

### CODE REVIEW

File: test/functional/filters/multilineexception/multiline_exception_test.go [hunk 1]
```diff
@@ -116,57 +113,18 @@ created by main.main
 		Entry("of V8 errors stack trace", jsV8Exception, nil),
 		Entry("of NodeJS services", nodeJSException, nil),
 		Entry("of GoLang services", goLangException, nil),
-		Entry("of single application NS to single pipeline", goLangException, func(framework *functional.CollectorFunctionalFramework) {
```
Comment (jcantrill) [inline test/functional/filters/multilineexception/multiline_exception_test.go:119]: Removed this as it tests the same functionality as the previous test

File: test/functional/filters/multilineexception/multiline_exception_test.go [hunk 2]
```diff
@@ -116,57 +113,18 @@ created by main.main
...
-				}).
-				WithMultineErrorDetection().
-				ToHttpOutput()
-			//LOG-2241
-			b.FromInput("multiline-log-ns").
-				Named("other").
-				WithMultineErrorDetection().
-				ToElasticSearchOutput()
-		}),
-		Entry("of multiple application NS source with multiple pipelines", goLangException, func(framework *functional.CollectorFunctionalFramework) {
```
Comment (jcantrill) [inline test/functional/filters/multilineexception/multiline_exception_test.go:152]: Removed as this is covered given input to pipeline to output are isolated so no different then other tests

**Review by cahartma — COMMENTED** [review]
/lgtm

**Pull request merged**